fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/rfc/RequestParameters.java

```diff
@@ -0,0 +1,239 @@
+// Copyright 2019 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.apirequest.rfc;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.ws.rs.core.PathSegment;
+
+/**
+ * Data object all raw pieces of data in an api request.
+ */
+public class RequestParameters {
```

Comment:
it's generally considered more appropriate to hold the nulls internally and optional wrap on return.  I don't remember if we discussed making an exception for this class or not.

Reply:
I suppose given the write to read ratio on this object, it actually saves cycles to wrap once.

Reply:
yeah that was the motivation. It's an immutable object that has no logic behind its internal state. Storing optionals over nulls just saves a little memory on read.